src/lib/connectors/local/local.ts

```diff
@@ -0,0 +1,300 @@
+import { getContentTypeData, isTextMediaType } from '../../utils/content-type';
+import { isFile, readFileAsync } from '../../utils/misc';
+import * as logger from '../../utils/logging';
+
+import {
+    IConnector, IConnectorBuilder,
+    IEvent, IFetchEnd, IScanEnd,
+    // INetworkData,
+    URL
+} from '../../types';
+// import { Requester } from '../utils/requester';
+import { Sonarwhal } from '../../sonarwhal';
+
+/*
+ * ------------------------------------------------------------------------------
+ * Defaults
+ * ------------------------------------------------------------------------------
+ */
+
+const debug: debug.IDebugger = d(__filename);
+
+const defaultOptions = {};
+
+class LocalConnector implements IConnector {
+    private _options;
+    private sonarwhal: Sonarwhal;
+    private _href: string;
+    private filesPattern: Array<string>;
+    private watcher;
+
+    public constructor(sonarwhal: Sonarwhal, config: object) {
+        this._options = Object.assign({}, defaultOptions, config);
+        this.filesPattern = this.getFilesPattern();
+        this.sonarwhal = sonarwhal;
+    }
+
+    /*
+     * ------------------------------------------------------------------------------
+     * Private methods
+     * ------------------------------------------------------------------------------
+     */
+    private getFilesPattern(): Array<string> {
+        const pattern = this._options.pattern;
+
+        if (!pattern) {
+            return ['**', '!.git/**'];
+        }
+
+        if (Array.isArray(pattern)) {
+            return pattern.length > 0 ? pattern : null;
+        }
+
+        return [pattern];
+    }
+
+    private async fetch(filePath: string) {
+        const rawContent = await fs.readFile(filePath);
+        const contentType = getContentTypeData(null, filePath, null, rawContent);
+        let content = '';
+        const extension = path.extname(filePath);
+
+        if (isTextMediaType(contentType.mediaType)) {
+            content = rawContent.toString(contentType.charset);
+        }
+
+        // Need to do some magic to create a fetch::end
+        const event: IFetchEnd = {
+            element: null,
+            request: null,
+            resource: url.format(getAsUri(filePath)),
+            response: {
+                body: {
+                    content,
+                    rawContent,
+                    rawResponse() {
+                        return Promise.resolve(rawContent);
+                    }
+                },
+                charset: contentType.charset,
+                headers: null,
+                hops: [],
+                mediaType: contentType.mediaType,
+                statusCode: 200,
+                url: filePath
+            }
+        };
+
+        if (extension === '.html') {
+            await this.sonarwhal.emitAsync('targetfetch::end', event);
+            // TODO: traverse the html.
+        } else if (extension === '.webmanifest') {
```

Comment:
`.json` is still a even more popular extension for the web app manifest file. :(

Are we planing to do some kind of detection?

Reply:
Maybe once we we have a Manifest parser we can do something more clever 😉

Reply:
So, yes, we will do some kind of detection. :P